docs/rst/tutorials/pkcs11_tutorial.rst

```diff
@@ -0,0 +1,168 @@
+.. _vulcanexus_pkcs11_tutorial:
+
+ROS 2 Security improved with Hardware Security Modules (HSM) using Vulcanexus Core
+==================================================================================
+
+.. contents::
+    :depth: 2
+    :local:
+    :backlinks: none
+
+Background
+----------
+
+`ROS 2 Security <https://docs.ros.org/en/humble/Concepts/About-Security.html>`_ is implemented by means of the `DDS Security specification <https://www.omg.org/spec/DDS-SECURITY/About-DDS-SECURITY/>`_.
+Specifically, when using Vulcanexus, security is implemented through `Fast DDS security plugins <https://fast-dds.docs.eprosima.com/en/latest/fastdds/security/security.html#security>`_.
+The security files required by the aforementioned specification are encapsulated within the ROS 2 security enclaves.
+This approach allows to set different security policies for different nodes, processes, users, or devices by defining different enclaves with their corresponding permissions.
+However, ROS 2 currently requires that the private key used to encrypt and decrypt the enclave information is also stored within the enclave as plain text, which might pose a security vulnerability.
+Hardware Security Modules (HSM) can be used in order to mitigate this vulnerability.
+And `PKCS #11 standard <http://docs.oasis-open.org/pkcs11/pkcs11-base/v2.40/os/pkcs11-base-v2.40-os.html>`_ provides an API to interact with these devices.
+Vulcanexus Core provides the required support for HSM devices so the private key, instead of being stored in the filesystem, it could be inside the HSM device.
+
+This tutorial provides step-by-step instructions to secure the communications within the ROS 2 talker/listener demo by the means of a HSM and leveraging Vulcanexus Core enhanced security capabilities.
+
+Prerequisites
+-------------
+
+This tutorial does not require a physical HSM device connected to the test environment.
+Instead, a software HSM emulator is going to be installed and used.
+Run the following commands to install the required packages (administrative privileges may be required):
+
+.. code-block:: bash
+
+    apt update
+    apt install --yes --no-install-recommends \
+      softhsm2 \
+      gnutls-bin
+
+Note that `softhsm2` package creates a new group called `softhsm`.
+In order to grant access to the HSM module a user must belong to this group.
+Run the following command (administrative privileges may be required):
```

Comment:
I'd clarify that this is not needed for the root user

Reply:
What I meant was more that root users do not need to be added to the `softhsm` group.

```
In order to grant access to the HSM module, non-root user must belong to this group (root users already have permission).
To add a user to the group, run the following command (administrative privileges may be required):
```

Reply:
Done!